core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/NamespaceChangeStreamListener.java

```diff
@@ -280,4 +282,21 @@ void storeNextEvent() {
       nsLock.writeLock().unlock();
     }
   }
+
+  /**
+   * If there is an unprocessed change event for a particular document ID, fetch it from the change
+   * stream listener without removing it.
+   *
+   * @return the latest unprocessed change event for the given document ID, or null if none exists.
+   */
+  public @Nullable ChangeEvent<BsonDocument> getUnprocessedEventForDocumentId(
```

Comment:
[minor?] Once this event is read, shouldn't it be considered processed? That was the logic behind clearing the map in `getEvents`– once they've been read, they have been processed. It seems like we should be removing them here.

Reply:
Agreed with Jason. This isn't a bug but it would cause us to keep doing work that's not needed.

Reply:
Ah, good catch, took me longer than it should have to wrap my head around this. I thought that because this is L2R we're not actually processing the event, and that we should wait for R2L to actually process it, but you're right that in any case, the event here is being effectively dropped or used to raise a conflict, so I'll fix this and clarify the comments here.

Reply:
Clarified the comments, and added a suggestion to the spec as well.